awx/api/views/__init__.py

```diff
@@ -2004,6 +2004,11 @@ class InventorySourceDetail(RelatedJobsPreventDeleteMixin, RetrieveUpdateDestroy
     model = models.InventorySource
     serializer_class = serializers.InventorySourceSerializer
 
+    def get_serializer_context(self):
+        context = super().get_serializer_context()
+        context.update(original=self.get_object())
```

Comment:
I don't know if this is the best way to do this. I needed the original object, not the object as represented by the request data. I didn't see a better way to get the object in the serializer without stuffing it in the context. This might be horrible, I'm not sure.

Reply:
I think a better place would be in the serializer `validate` method (or `update` like now) before the `super()` call. The criteria for checking for a change goes get a little messy. For every field in `attrs`, if it's not in the allowed list for changing (given it's a constructed source type) then if `attrs` value differs from the object, then you want to throw a validation message.

Reply:
This is effectively what I do in the serializer. But I needed access to the original object as it exists in the database when we reach the serializer.

The `obj` that gets passed to `update()` seems to already have the request data in it. So I can't check if it's different, because it never will be.

So here I'm sending along the original object in the serializer context so that I have something to compare the new object (in `obj`) to.